### PR TITLE
OCPBUGS-15852: Single node cannot be installed if etcd appears in the hostname

### PR DESCRIPTION
**Pull request by eranco74**
Add filter by etcd container name regular expression pattern before the grep

https://issues.redhat.com/browse/OCPBUGS-15852

### CODE REVIEW

File: data/data/bootstrap/bootstrap-in-place/files/usr/local/bin/bootstrap-in-place.sh [hunk 1]
```diff
@@ -16,7 +16,7 @@ if [ ! -f stop-etcd.done ]; then
   echo "Stop etcd static pod by moving the manifest"
   mv /etc/kubernetes/manifests/etcd-member-pod.yaml /etc/kubernetes || echo "already moved etcd-member-pod.yaml"
 
-  until ! crictl ps | grep etcd; do
+  until ! crictl ps --name etcd | grep etcd; do
```
Comment (omertuc) [inline data/data/bootstrap/bootstrap-in-place/files/usr/local/bin/bootstrap-in-place.sh:19]: `--name etcd` might still capture `etcd` in container names which might contain the hostname, maybe we should work with structured output instead:

```suggestion
  until sudo crictl ps --state running -o json | jq -e '[.containers[] | select(.metadata.name == "etcd")] | length == 0'; do
```

Comment (eranco74) [reply data/data/bootstrap/bootstrap-in-place/files/usr/local/bin/bootstrap-in-place.sh]: I'm sure that the container name during bootstrap is just etcd, yet this jq seems more robust

Comment (omertuc) [reply data/data/bootstrap/bootstrap-in-place/files/usr/local/bin/bootstrap-in-place.sh]: The etcd container name is just etcd, but other containers might be running and might have the hostname in their name

File: data/data/bootstrap/bootstrap-in-place/files/usr/local/bin/bootstrap-in-place.sh [hunk 2]
```diff
@@ -16,7 +16,7 @@ if [ ! -f stop-etcd.done ]; then
   echo "Stop etcd static pod by moving the manifest"
   mv /etc/kubernetes/manifests/etcd-member-pod.yaml /etc/kubernetes || echo "already moved etcd-member-pod.yaml"
 
-  until ! crictl ps | grep etcd; do
+  until sudo crictl ps --state running -o json | jq -e '[.containers[] | select(.metadata.name == "etcd")] | length == 0' > /dev/null; do
```
Comment (omertuc) [inline data/data/bootstrap/bootstrap-in-place/files/usr/local/bin/bootstrap-in-place.sh:19]: Sorry didn't mean to suggest sudo

**Comment by sadasu** [comment]
/retitle OCPBUGS-15852: Single node cannot be installed if etcd appears in the hostname

**Pull request merged**